### ISSUE TITLE
Autocomplete not working

### ISSUE THREAD

**Issue by Julian-Th** [sergeyt/meteor-typeahead#135]
Hallo,

for some reason I can't get your autocomplete package to work. Would you care to glance over my html and the associated .js file?

**HTML INPUT FORM** (in a template called "modalAddCollaborators"):

```<input class="form-control typeahead" type="text" id="collaboratorName" placeholder="add a collaborator  ..." data-source="courses" autocomplete="off" spellcheck="off">```

**JS file:**

```javascript 
Template.modalAddCollaborators.rendered = function() {
    Meteor.typeahead.inject();
};

Template.modalAddCollaborators.courses = function(){
	return Courses.find().fetch().map(function(it){ return it.author; });
};
```

What did I miss?

Thanks a lot for your help!

Julian

**Comment by sergeyt** [comment]
@Julian-Th your code looks fine at first glance. Don't forget to include CSS styles (e.g. take it from demo app). Also try to change `cources` function returning objects instead of strings, e.g.:

```js
Template.modalAddCollaborators.courses = function(){
    return Courses.find().fetch().map(function(it){ return {value: it.author, id: it._id}; });
};
```

**Comment by Julian-Th** [comment]
Sergey, I really appreciate your instant response! Unfortunately I haven't managed to get it to work yet. I copied the CSS style into modalAddCollaborators.css and changed the return value to an object. 
Is there anything else you can think of that might be causing the package to not work properly?

**Comment by Julian-Th** [comment]
Sorry for these basic questions. I'm very new to meteor.js. Maybe you find it easier to give a response by looking at the git repo?

https://github.com/Julian-Th/crowducate-platform/tree/feature/courseEditRights/client/templates/shared

**Comment by sergeyt** [comment]
@Julian-Th I guess issue with modal dialog. Try put your typeahead input outside the modal dialog just to check that it works. Again CSS styles are required. Also try to increase z-index for typeahead dropdown.

**Comment by Julian-Th** [comment]
Ok thanks, sounds like a good plan. WIll check that later and report back! Again, thank you very much for your support!

**Comment by Julian-Th** [comment]
Sergey, even outside the modal dialog the autocomplete doesn't seem to work. :(
Would you care to take a look at the repo I linked above? I really don't know what else to try and I have a deadline of a MOOC project on Dec. 30th... :-/

The autocomplete implementation of the modal dialog is in modalAddCollaborators.html, modalAddCollaborators.js and modalAddCollaborators.css.


Many thanks!!

Julian

**Comment by sergeyt** [comment]
@Julian-Th I've looked under debugger your app. Here the step to get typeahead package working:

1. Include required typeahead [CSS styles](https://github.com/sergeyt/meteor-typeahead/blob/master/demo/style.css), not demo app styles!

2. Move `courses` function into template helpers, i.e.:

```js
Template.modalAddCollaborators.helpers({
	courses: function() {
		return Courses.find().fetch().map(function(it){ return it.author; });
	},
	addedCollaborators: function () {
		return Courses.find().fetch();
	}
});
```

This second step is required now for the latest package version v0.11.1_6. Actually this is a regression happened after upgrade to meteor 1.2 (see ac5376f567dc824daf496fad64f637c48f5fb3df), i.e. old style helpers can not be detected by the package (`resolve_helper` function only uses `template.__helpers`). I'll try to fix this issue soon.

**Comment by sergeyt** [comment]
@Julian-Th FYI I've just published v0.11.1_7 with the fix for old-style helpers. So to get the typeahead working in your app you can update the package to v0.11.1_7 and include required [CSS styles](https://github.com/sergeyt/meteor-typeahead/blob/master/demo/style.css).

**Comment by Julian-Th** [comment]
It's finally working! I can't thank you enough, Sergey! :)) Awesome support.
